src/js/components/PodSpecView.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import PodSpec from '../structs/PodSpec';
+import PodContainerSpecView from './PodContainerSpecView';
+import Icon from './Icon';
+
+const METHODS_TO_BIND = [
+];
+
+class PodSpecView extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getEnvironmentDetails() {
+    let {spec} = this.props;
+    let environment = Object.assign({}, spec.getEnvironment());
```

Comment:
This could probably be re-written to something like this:
```js
let environment = spec.getEnvironment();

if (Object.keys(environment).length === 0) {
  return null;
}

let hash = Object.keys(environment).reduce(function (memo, key) {
  let value = environment[key];
  if (typeof value === 'object' && !Array.isArray(value) && value !== null) {
    memo[key] = (
      <span>
        <Icon
          className="icon-margin-right"
          color="white"
          family="mini"
          id="key"
          size="mini" /> {value.secret}
      </span>
    );
  }

  return memo;
}, {});
```